### PR TITLE
Log: New premium barrier API call errors

### PR DESCRIPTION
**Pull request by andygout**
Produces locally (as an example, to which I added `err=${err}` to see what that looked like): `event=failureGettingUpgradeOption message=Cannot read property 'billedMonthly' of undefined, err=TypeError: Cannot read property 'billedMonthly' of undefined level=error`

### CODE REVIEW

File: server/controllers/premium-new.js [hunk 1]
```diff
@@ -67,6 +70,7 @@ const render = (req, res) => {
 			.then(getBarrierRenderData)
 			.catch(err => {
 				if (!err.isCancellation) {
+					logger.error(`event=failureValidatingSession message=${err.message}`, err);
```
Comment (ironsidevsquincy) [inline server/controllers/premium-new.js]: no need for explicitly ouputting `err.message`, n-logger will already do this if an Error object is its own argument.
also, use an object for fields

```
logger.error({event: 'failureValidatingSession' }, err);
```

Comment (ironsidevsquincy) [inline server/controllers/premium-new.js]: And I think general standard is for the `event` to be capitalised, `FAILURE_VALIDATING_SESSION`

Comment (andygout) [inline server/controllers/premium-new.js]: Is it possible err will be output as `[Object object]`?

Comment (ironsidevsquincy) [inline server/controllers/premium-new.js]: no. n-logger handles Error objects. https://github.com/Financial-Times/n-logger#api

**Pull request merged**